### PR TITLE
Add UCS Error handeling and negative unitTesting

### PR DESCRIPTION
**Pull request by tannoa2**
- add an http error factory
- add error handeling for the UCS server
- add flake8 to the requirement.txt
- add negative unit testing

### CODE REVIEW

File: controllers/default_controller.py [hunk 1]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
             units = []
-            components = handle.query_children(in_dn="sys", class_id=x["ciscoXmlName"])
-            for y in components:
-                subElement = {"relative_path": "/" + (vars(y))["dn"]}
-                units.append(subElement)
-            finalObjs[x["humanReadableName"]] = units
-            handle.logout()
+            try:
+                components = handle.query_children(in_dn="sys", class_id=x["ciscoXmlName"])
+            except UcsException, e:
```
Comment (geoff-reid) [inline controllers/default_controller.py]: This needs to be fixed.

File: controllers/default_controller.py [hunk 2]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
+                    for y in components:
+                        subElement = {"relative_path": "/" + (vars(y))["dn"]}
+                        units.append(subElement)
+                    finalObjs[x["humanReadableName"]] = units
+                    handle.logout()
+                else:
+                    return "Couldn't fetch" + x["ciscoXmlName"], "", 500
         return finalObjs
+    else:
+        return 'Forbiden', "", 403
```
Comment (geoff-reid) [inline controllers/default_controller.py]: Need to logout before return.

File: controllers/default_controller.py [hunk 3]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
+                return 'Inrternal Server Error', e.error_code, 500
+            else:
+                if(type(components) == list):
+                    for y in components:
+                        subElement = {"relative_path": "/" + (vars(y))["dn"]}
+                        units.append(subElement)
+                    finalObjs[x["humanReadableName"]] = units
+                    handle.logout()
+                else:
+                    return "Couldn't fetch" + x["ciscoXmlName"], "", 500
```
Comment (geoff-reid) [inline controllers/default_controller.py]: Need to logout before return.

File: controllers/default_controller.py [hunk 4]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
+                    server["macs"] = []
+                    macs = handle.query_children(in_dn=x.dn, class_id='PciEquipSlot')
+                    for y in macs:
+                        server["macs"].append(y.mac_left)
+                        server["macs"].append(y.mac_right)
+                    data.append(server)
+                    handle.logout()
+                    return data
+            else:
+                return "Couldn't fetch" + computeRackUnit + "s", "", 500
```
Comment (geoff-reid) [inline controllers/default_controller.py]: Need to logout before return.

File: controllers/default_controller.py [hunk 5]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
+
+    else:
+        return 'Forbiden', "", 403
```
Comment (geoff-reid) [inline controllers/default_controller.py]: Need to logout before return.

File: controllers/default_controller.py [hunk 6]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
-            components = handle.query_children(in_dn="sys", class_id=x["ciscoXmlName"])
-            for y in components:
-                subElement = {"relative_path": "/" + (vars(y))["dn"]}
-                units.append(subElement)
-            finalObjs[x["humanReadableName"]] = units
-            handle.logout()
+            try:
+                components = handle.query_children(in_dn="sys", class_id=x["ciscoXmlName"])
+            except UcsException, e:
+                return 'Inrternal Server Error', e.error_code, 500
```
Comment (geoff-reid) [inline controllers/default_controller.py]: Need to logout before return.

File: controllers/default_controller.py [hunk 7]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
-                server["macs"].append(y.mac_left)
-                server["macs"].append(y.mac_right)
-            data.append(server)
-            handle.logout()
-        return data
+        try:
+            computeRackUnit = handle.query_children(in_dn="sys", class_id="computeRackUnit")
+            print computeRackUnit
+        except UcsException, e:
+            return 'Inrternal Server Error', e.error_code, 500
```
Comment (geoff-reid) [inline controllers/default_controller.py]: Need to logout before return.

File: controllers/default_controller.py [hunk 8]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
     if handle.login():
-        elements = handle.query_children(in_dn=identifier)
-        for element in elements:
-            data.append(reduce(element.__dict__))
-        handle.logout()
-        return data
+        try:
+            elements = handle.query_children(in_dn=identifier)
+        except UcsException, e:
+            return 'Inrternal Server Error', e.error_code, 500
```
Comment (geoff-reid) [inline controllers/default_controller.py]: Need to logout before return.

File: controllers/default_controller.py [hunk 9]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
+        else:
+            if (type(elements) == list):
+                for element in elements:
+                    data.append(reduce(element.__dict__))
+                handle.logout()
+                return data
+            else:
+                return "Couldn't fetch " + identifier, "", 500
```
Comment (geoff-reid) [inline controllers/default_controller.py]: Need to logout before return.

File: controllers/default_controller.py [hunk 10]
```diff
@@ -19,44 +36,75 @@ def systemGetAll(host=None, user=None, password=None):
...
+    else:
+        return 'Forbiden', "", 403
```
Comment (geoff-reid) [inline controllers/default_controller.py]: Need to logout before return.

**Comment by tannoa2** [comment]
@RackHD/veyron

**Comment by tannoa2** [comment]
@geoff-reid  Comments has been addressed

**Pull request merged**